File: app/src/main/java/com/example/myfakequotesapp/api/ApiFactory.java
package com.example.myfakequotesapp.api;

import android.app.DownloadManager;
import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.myfakequotesapp.BuildConfig;
import com.example.myfakequotesapp.pojo.Example;
import com.example.myfakequotesapp.pojo.Result;
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;

import java.util.List;
import java.util.stream.Collectors;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import static android.content.ContentValues.TAG;

public class ApiFactory {
    private static ApiFactory apiFactory;
    private static Retrofit retrofit;
    private static final String BASE_URL = "https://imdb-api.com/en/API/SearchTitle/k_w1rp6vvu/";

    private ApiFactory() {
        retrofit = new Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .baseUrl(BASE_URL)
                .build();
    }

    public LiveData<Example> getExample(String searchTitle) {
        MutableLiveData<Example> liveExamplesData = new MutableLiveData<>();
        //List
        apiFactory.getApiService().getExample(searchTitle).enqueue(new Callback<Example>() {
            @Override
            public void onResponse(Call<Example> call, Response<Example> response) {
                if (response.isSuccessful() && response.body() != null) {
                    liveExamplesData.setValue(response.body());
                    Log.d(TAG, "INFO: " + response);
                }
            }

            @Override
            public void onFailure(Call<Example> call, Throwable t) {
                Log.d(TAG, "ERROR: " + t.getMessage());
            }
        });
        return liveExamplesData;
    }

    public static ApiFactory getInstance() {
        if (apiFactory == null) {
            apiFactory = new ApiFactory();
        }
        return apiFactory;
    }

    public ApiService getApiService() {
        return retrofit.create(ApiService.class);
    }
}

File: app/src/main/java/com/example/myfakequotesapp/api/ApiFactory.kt
package com.example.myfakequotesapp.api

import android.app.DownloadManager
import android.content.ContentValues
import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.myfakequotesapp.BuildConfig
import com.example.myfakequotesapp.pojo.Example
import com.example.myfakequotesapp.pojo.Result
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import java.util.stream.Collectors
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import android.content.ContentValues.TAG


//import com.example.myfakequotesapp.BuildConfig;
class ApiFactory private constructor() {
    fun getExample(searchTitle: String?): LiveData<Example?> {
        val liveExamplesData = MutableLiveData<Example?>()
        //List
        apiFactory!!.apiService.getExample(searchTitle).enqueue(object : Callback<Example?> {
            override fun onResponse(call: Call<Example?>, response: Response<Example?>) {
                if (response.isSuccessful && response.body() != null) {
                    liveExamplesData.value = response.body()
                    Log.d(ContentValues.TAG, "INFO: $response")
                }
            }

            override fun onFailure(call: Call<Example?>, t: Throwable) {
                Log.d(ContentValues.TAG, "ERROR: " + t.message)
            }
        })
        return liveExamplesData
    }

    val apiService: ApiService
        get() = retrofit.create(ApiService::class.java)

    companion object {
        private var apiFactory: ApiFactory? = null
        private lateinit var retrofit: Retrofit
        private const val BASE_URL = "https://imdb-api.com/en/API/SearchTitle/k_w1rp6vvu/"
        val instance: ApiFactory?
            get() {
                if (apiFactory == null) {
                    apiFactory = ApiFactory()
                }
                return apiFactory
            }
    }

    init {
        retrofit = Retrofit.Builder()
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .baseUrl(BASE_URL)
            .build()
    }
}